### PR TITLE
Add AWS support for neural chat multi-node fine-tuning

### PR DESCRIPTION
**Pull request by louie-tsai**
## Type of Change

Add AWS support for neural chat multi-node fine-tuning

## Description

1. new AWS inbound rule
2. give custom SSH server port during docker build

## Expected Behavior & Potential Risk

works fine on AWS m7i SPR instances

## How has this PR been tested?

AWS m7i.8xlarge instances

## Dependency Change?
no

### CODE REVIEW

File: intel_extension_for_transformers/neural_chat/examples/finetuning/instruction/README.md [hunk 1]
```diff
@@ -402,13 +402,40 @@ mpirun -f nodefile -n 16 -ppn 4 -genv OMP_NUM_THREADS=56 python3 finetune_clm.py
...
+ex : using 2345 as the CUSTOM_PORT  
+```bash
+docker build --build-arg UBUNTU_VER=22.04 --build-arg SSHD_PORT=2345 -f intel-extension-for-transformers/intel_extension_for_transformers/neural_chat/docker/Dockerfile -t ${IMAGE_NAME}:${IMAGE_TAG} . --target cpu
+```  
+
+### Add one AWS inbound rule for distributed training
+Allow all network traffic inside the cluster, so that distributed training runs unencumbered.   
+AWS provides a safe and convenient way to do this with security groups. We just need to create a security group that allows all traffic from instances configured with that same security group and make sure to attach it to all instances in the cluster.  
+Here's how my setup looks.  
+<img width="655" alt="image" src="https://github.com/intel-ai-tce/intel-extension-for-transformers/assets/21761437/baf98fdd-14ff-457c-afed-a387b164c2df">
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/examples/finetuning/instruction/README.md:429]: better to add the assets under itrex https://github.com/intel/intel-extension-for-transformers/tree/main/intel_extension_for_transformers/neural_chat/assets

File: intel_extension_for_transformers/neural_chat/examples/finetuning/instruction/README.md [hunk 2]
```diff
@@ -402,13 +402,40 @@ mpirun -f nodefile -n 16 -ppn 4 -genv OMP_NUM_THREADS=56 python3 finetune_clm.py
...
+Allow all network traffic inside the cluster, so that distributed training runs unencumbered.   
+AWS provides a safe and convenient way to do this with security groups. We just need to create a security group that allows all traffic from instances configured with that same security group and make sure to attach it to all instances in the cluster.  
+Here's how my setup looks.  
+<img width="655" alt="image" src="https://github.com/intel-ai-tce/intel-extension-for-transformers/assets/21761437/baf98fdd-14ff-457c-afed-a387b164c2df">  
+Users could also refer to [a huggingface blog](https://huggingface.co/blog/intel-sapphire-rapids) for more details.
+
+### Same Instructions as Multi-node Fine-tuning in Xeon SPR session
+Please follow previous Multi-node Fine-tuning in Xeon SPR session with the docker image and AWS inbound rule changes.  
+For the IPs in nodefile, please **use private IP instead of public IP**.  
+<img width="710" alt="image" src="https://github.com/intel-ai-tce/intel-extension-for-transformers/assets/21761437/9aad158d-4053-4bc6-895d-bca0f1b2219c">
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/examples/finetuning/instruction/README.md:435]: Same as above

**Review by jiafuzha — APPROVED** [review]
LGTM

File: intel_extension_for_transformers/neural_chat/examples/finetuning/instruction/README.md (hunk 1 above)
Comment (louie-tsai) [reply intel_extension_for_transformers/neural_chat/examples/finetuning/instruction/README.md]: addressed it accordingly.

File: intel_extension_for_transformers/neural_chat/examples/finetuning/instruction/README.md (hunk 2 above)
Comment (louie-tsai) [reply intel_extension_for_transformers/neural_chat/examples/finetuning/instruction/README.md]: addressed it accordingly.

**Comment by louie-tsai** [comment]
@hshen14 
Addressed the comments accordingly. please let me know any further change needed.

**Review by lvliang-intel — APPROVED** [review]

**Comment by hshen14** [comment]
Thanks @louie-tsai  for your contribution!

**Pull request merged**